frontend/pages/queries/details/components/QueryReport/QueryReportTableConfig.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/prop-types */
+// disable this rule as it was throwing an error in Header and Cell component
+// definitions for the selection row for some reason when we dont really need it.
+import React from "react";
+
+import {
+  CellProps,
+  Column,
+  ColumnInstance,
+  ColumnInterface,
+  HeaderProps,
+  TableInstance,
+} from "react-table";
+
+import DefaultColumnFilter from "components/TableContainer/DataTable/DefaultColumnFilter";
+import HeaderCell from "components/TableContainer/DataTable/HeaderCell/HeaderCell";
+
+import { humanHostLastSeen } from "utilities/helpers";
+
+type IHeaderProps = HeaderProps<TableInstance> & {
+  column: ColumnInstance & IDataColumn;
+};
+
+type ICellProps = CellProps<TableInstance>;
+
+interface IDataColumn extends ColumnInterface {
+  title?: string;
+  accessor: string;
+}
+
+const _unshiftHostname = (headers: IDataColumn[]) => {
+  const newHeaders = [...headers];
+  const displayNameIndex = headers.findIndex(
+    (h) => h.id === "host_display_name"
+  );
+  if (displayNameIndex >= 0) {
+    // remove hostname header from headers
+    const [displayNameHeader] = newHeaders.splice(displayNameIndex, 1);
+    // reformat title and insert at start of headers array
+    newHeaders.unshift({ ...displayNameHeader, title: "Host" });
+  }
+  // TODO: Remove after v5 when host_hostname is removed rom API response.
+  const hostNameIndex = headers.findIndex((h) => h.id === "host_hostname");
+  if (hostNameIndex >= 0) {
+    newHeaders.splice(hostNameIndex, 1);
+  }
+  // end remove
+  return newHeaders;
+};
+
+const generateResultsTableHeaders = (results: any[]): Column[] => {
+  /* Results include an array of objects, each representing a table row
+  Each key value pair in an object represents a column name and value
+  To create headers, use JS set to create an array of all unique column names */
+  const uniqueColumnNames = Array.from(
+    results.reduce(
+      (s, o) => Object.keys(o).reduce((t, k) => t.add(k), s),
```

Comment:
This could be clearer. The single var names are tricky to reason about when combined with the nested `reduce`s. Maybe:

`s` –> `uniqueColumns`
`o` –> `resultColumn`

and then add each key of `resultColumn` to `uniqueColumns` before returning `uniqueColumns`. You might do a `return new Set.from(...uniqueColumns, ...Object.keys(resultColumn)`.

Feels like this is getting to be a lot of nested `O(n)`s and might be introducing some performance issues if the number of columns gets to be nontrivial. May be worth thinking about a simpler way to do this.

Reply:
Just the variable renames would be nice. Not necessary for now though.